### PR TITLE
Ruby: Support for page iteration tests

### PR DESCRIPTION
**Pull request by landrito**

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/ruby_test_library.baseline [hunk 1]
```diff
@@ -183,6 +183,58 @@ describe Library::V1::LibraryServiceClient do
...
+    LibraryServiceClient = Library::V1::LibraryServiceClient
+    ListShelvesResponse = Google::Example::Library::V1::ListShelvesResponse
+    Shelf = Google::Example::Library::V1::Shelf
+
+    it 'invokes list_shelves without error' do
+      # Create expected grpc response
+      next_page_token = ""
+      shelves_element = Shelf.new
+      shelves = [shelves_element]
+      expected_response = ListShelvesResponse.new(next_page_token: next_page_token, shelves: shelves)
```
Comment (swcloud) [inline src/test/java/com/google/api/codegen/testdata/ruby_test_library.baseline]: It looks like we're testing the response which is paged?

Comment (landrito) [inline src/test/java/com/google/api/codegen/testdata/ruby_test_library.baseline]: Good point let me add that!

Comment (landrito) [inline src/test/java/com/google/api/codegen/testdata/ruby_test_library.baseline]: Done.

File: src/main/resources/com/google/api/codegen/ruby/test.snip [hunk 2]
```diff
@@ -144,6 +146,8 @@
       {@methodCallWithResponse(test)}
 
       @# Verify the response
+      assert(response.instance_of?(Google::Gax::PagedEnumerable))
+      assert_equal(expected_response, response.page.response)
```
Comment (swcloud) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Maybe add

assert_nil(response.page.next_page)

Comment (landrito) [inline src/main/resources/com/google/api/codegen/ruby/test.snip]: Done using `PagedEnumerable#next_page` rather than `PagedEnumerable::Page#next_page`.

**Pull request merged**